### PR TITLE
feat: create v-timeline

### LINKED ISSUE #1340: [Feature Request] v-timeline
I am looking for something like timeline which i can implement in my web app

Here is the code-pen of material design timeline - https://codepen.io/samerpik/pen/OVKyVZ
Looking for somewhat same for VuetifyJS

Timeline in Angular - http://rp.js.org/angular-timeline/example/index.html

### PR DESCRIPTION
**Pull request by johnleider**
## Description
This is for timeline component for vuetify.

## Motivation and Context
resolves #1340

## How Has This Been Tested?
In playground

## Todo

- [x] tests

## Markup:
```vue
<template>
  <v-app>
    <v-content>
      <v-container fluid fill-height>
        <v-layout align-center justify-center>
          <v-flex>
          Basic
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i">
              <v-card>
                <v-card-text>
                  Hello this is text.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Dark
          <div style="padding: 1rem; background: #303030">
            <v-timeline dark>
              <v-timeline-item v-for="(n,i) in items" :key="i">
                <v-card>
                  <v-card-text>
                    Hello this is text.
                  </v-card-text>
                </v-card>
              </v-timeline-item>
            </v-timeline>
          </div>

          Small dot
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i" small>
              <v-card>
                <v-card-text>
                  Hello this is text.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Icons
          <v-timeline>
            <v-timeline-item icon="location_on" v-for="(n,i) in items" :key="i" small>
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Color
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i" color="orange">
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Icon color
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i" icon="location_on" icon-color="red" color="transparent">
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Position
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i" :left="i < 2" :right="i >= 2">
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Slot - opposite
          <v-timeline>
            <v-timeline-item v-for="(n,i) in items" :key="i">
              <div slot="opposite">This is opposite text</div>
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Dense
          <v-timeline dense>
            <v-timeline-item v-for="(n,i) in items" :key="i">
              <v-card>
                <v-card-text>
                  Hello world.
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>

          Card not necessary
          <v-timeline dense>
            <v-timeline-item v-for="(n,i) in items" :key="i">
              <v-alert :value="true" color="warning">
                Hello world.
              </v-alert>
            </v-timeline-item>
          </v-timeline>

          Full use case
          <v-timeline>
            <v-timeline-item v-for="(item, i) in items" :key="i" :icon="item.icon" :color="item.color">
              <div slot="opposite">{{ item.date }}</div>
              <v-card>
                <v-card-text>
                  {{ item.text }}
                </v-card-text>
              </v-card>
            </v-timeline-item>
          </v-timeline>
          </v-flex>
        </v-layout>
      </v-container>
    </v-content>
  </v-app>
</template>

<script>
export default {
  data: () => ({
    items: [
      {
        date: '2018-01-01',
        title: 'Hello world',
        text: 'This is text.',
        color: 'primary',
        icon: 'location_on'
      },
      {
        date: '2018-01-03',
        title: 'Hello world',
        text: 'This is text.',
        color: 'orange',
        icon: 'home'
      },
      {
        date: '2018-01-05',
        title: 'Hello world',
        text: 'This is text.',
        color: 'purple',
        icon: 'favorite'
      },
      {
        date: '2018-01-07',
        title: 'Hello world',
        text: 'This is text.',
        color: 'green',
        icon: 'explore'
      }
    ]
  })
}
</script>
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by Seraf** [comment]
Hello, any news on this PR @abhaywawale ?
This component is really exciting 👍

**Comment by abhaywawale** [comment]
@Seraf it is planned for Front-end pack. You can see [here](https://github.com/vuetifyjs/vuetify/milestone/8).

File: src/components/VTimeline/VTimeline.ts [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+
+// Types
 import Vue, { VNode } from 'vue'
+import mixins from '../../util/mixins'
+
+// Mixins
+import Themeable from '../../mixins/themeable'
 
-export default Vue.extend({
+/* @vue/component */
```
Comment (KaelWD) [inline src/components/VTimeline/VTimeline.ts]: wrong line, needs to be directly above the start of the object

File: src/stylus/components/_timeline.styl [hunk 2]
```diff
@@ -3,10 +3,24 @@
 
 v-timeline($material)
   &:before
-    background: $material.dividers
+    background: $material.timeline.active
+
+  .v-timeline-item
```
Comment (KaelWD) [inline src/stylus/components/_timeline.styl]: timeline-item should have its own theme function

File: src/stylus/components/_timeline.styl [hunk 3]
```diff
@@ -3,10 +3,24 @@
...
+    &__dot
+      background: $material.cards
+
+    &:nth-child(odd):not(.v-timeline-item--right), &--left
+      .v-timeline-item__body:before
+        border-color: transparent transparent transparent black
```
Comment (KaelWD) [inline src/stylus/components/_timeline.styl]: Is black right for dark theme too? If it doesn't use the $material variable it shouldn't be in this section

File: src/stylus/components/_timeline.styl [hunk 4]
```diff
@@ -3,10 +3,24 @@
...
+
+    &:nth-child(even):not(.v-timeline-item--left), &--right
+      .v-timeline-item__body:before
+        border-color: transparent black transparent transparent
 
 theme(v-timeline, "v-timeline")
 
+$wedge-size := 10px
```
Comment (KaelWD) [inline src/stylus/components/_timeline.styl]: Should be in the global _variables file

**Pull request merged**